### PR TITLE
Update minimum supported PHP version, add allowed plugins

### PR DESCRIPTION
**Pull request by internalsystemerror**
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | no
| New Feature   | no
| RFC           | no
| QA            | yes

### Description

This PR:
- Updates the minimum supported PHP version to 7.4
- Adds allowed plugins to fix the issues brought up by https://github.com/mezzio/mezzio-session-ext/pull/32
- Updates `phpunit/phpunit` to `~2.4.0` as the XSD is required by CI

### CODE REVIEW

**Comment by internalsystemerror** [comment]
Psalm will fail with the following still, but I'm unsure right now how to address them:

>ERROR: MixedArgumentTypeCoercion - src/PhpSessionPersistence.php:112:28 - Argument 1 of Mezzio\Session\Session::__construct expects array<string, mixed>, but parent type array<array-key, mixed> provided (see https://psalm.dev/194)
        return new Session($_SESSION ?? [], $sessionId);


>ERROR: MixedArgumentTypeCoercion - src/PhpSessionPersistence.php:159:36 - Argument 1 of Mezzio\Session\Session::__construct expects array<string, mixed>, but parent type array<array-key, mixed> provided (see https://psalm.dev/194)
            $session = new Session($session->toArray(), $this->generateSessionId());

>The type of $session->toArray() is sourced from here - vendor/mezzio/mezzio-session/src/SessionInterface.php:12:32
    public function toArray(): array;

We may need to add better types to `mezzio/mezzio-session`.

**Comment by Ocramius** [comment]
Baseline additions would suffice at first, perhaps?

File: src/ConfigProvider.php [hunk 1]
```diff
@@ -24,8 +22,8 @@ public function getDependencies(): array
                 SessionPersistenceInterface::class => PhpSessionPersistence::class,
 
                 // Legacy Zend Framework aliases
-                LegacySessionPersistenceInterface::class => SessionPersistenceInterface::class,
-                LegacyPhpSessionPersistence::class       => PhpSessionPersistence::class,
+                'use Zend\Expressive\Session\SessionPersistenceInterface' => SessionPersistenceInterface::class,
```
Comment (samsonasik) [inline src/ConfigProvider.php]: `use ` here is a typo copy paste ?

Comment (internalsystemerror) [reply src/ConfigProvider.php]: ah arse, good spot! will fix tonight!

**Review by Ocramius — APPROVED** [review]
LGTM 👍

File: src/ConfigProvider.php (hunk 1 above)
Comment (Ocramius) [reply src/ConfigProvider.php]: Eagle eyes 😁

**Pull request merged**